#OpenXT:Ask for password only while connecting first time.
#       This is nothing but 'Remember Password' option.
--- a/src/devices/wifi/nm-device-wifi.c
+++ b/src/devices/wifi/nm-device-wifi.c
@@ -2273,8 +2273,14 @@ supplicant_connection_timeout_cb (gpoint
 		 * dialogs, just retry or fail, and if we never connect the user can
 		 * fix the password somewhere else.
 		 */
-		if (nm_settings_connection_get_timestamp (nm_act_request_get_settings_connection (req), &timestamp))
-			new_secrets = !timestamp;
+		nm_settings_connection_get_timestamp (nm_act_request_get_settings_connection (req), &timestamp);
+
+		if (timestamp) {
+			new_secrets = FALSE;
+			nm_log_warn (LOGD_DEVICE | LOGD_WIFI,
+					"Activation (%s/wireless): previously connected (timestamp %llu)",
+					nm_device_get_iface (device));
+		}
 
 		if (handle_auth_or_fail (self, req, new_secrets) == NM_ACT_STAGE_RETURN_POSTPONE)
 			_LOGW (LOGD_DEVICE | LOGD_WIFI, "Activation: (wifi) asking for new secrets");
